app/projectsmodel.cpp

```diff
@@ -30,34 +30,61 @@ ProjectModel::ProjectModel(const QString &dataDir, QObject* parent)
 ProjectModel::~ProjectModel() {}
 
 void ProjectModel::findProjectFiles() {
-    addProjectsFromPath(mDataDir);
+    QStringList entryList = QDir(mDataDir).entryList(QDir::NoDotAndDotDot | QDir::Dirs);
+    for (QString folderName: entryList) {
+        addProjectFromPath(mDataDir + "/" + folderName);
+    }
+    std::sort(mProjectFiles.begin(), mProjectFiles.end());
 }
 
-void ProjectModel::addProjectsFromPath(QString path)
+void ProjectModel::addProjectFromPath(QString path)
 {
+    if (path.isEmpty()) return;
+
     QDirIterator it(path, QStringList() << QStringLiteral("*.qgs"), QDir::Files, QDirIterator::Subdirectories);
     QSet<QString> projectFilePaths;
 
     for (ProjectFile projectFile: mProjectFiles) {
         projectFilePaths << projectFile.path;
     }
 
+
+    QList<ProjectFile> foundProjects;
     while (it.hasNext())
     {
         it.next();
         ProjectFile projectFile;
         projectFile.name = it.fileName().remove(".qgs");
         projectFile.path = it.filePath();
+        QDir projectDir(path);
+        projectFile.folderName = projectDir.dirName();
         QFileInfo fileInfo(it.filePath());
         QDateTime created = fileInfo.created();
         projectFile.info = QString(created.toString());
+        projectFile.isValid = true;
 
         if (!projectFilePaths.contains(projectFile.path))
-            mProjectFiles.append(projectFile);
+            foundProjects.append(projectFile);
 
         qDebug() << "Found QGIS project: " << it.filePath();
     }
-    std::sort(mProjectFiles.begin(), mProjectFiles.end());
+
+    ProjectFile project;
+    if (!foundProjects.isEmpty()) {
+        project = foundProjects.at(0);
+        if (foundProjects.length() > 1) {
+            project.info = "invalid project";
+            project.isValid = false;
+        }
```

Comment:
why do we set the first project as invalid?

Reply:
Projects files are read per folder. If there is more than one *.qgs project file (*.qgz is still not supported yet), the first ProjectFile is taken, marked as invalid and added to projectList, others are ignored. Its because of avoiding duplicity of projects.